### PR TITLE
[dagster_model] DbtProject

### PR DESCRIPTION
**Pull request by alangenfeld**
DagsterModel -> dagster_model

## How I Tested These Changes

existing coverage

### CODE REVIEW

**Comment by alangenfeld** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22332?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#22332** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22332?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#22330** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22330?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22043** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22043?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22260** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22260?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 1]
```diff
@@ -220,7 +221,8 @@ def __init__(
             dependencies_path.exists() or packages_path.exists()
         ) and not packages_install_path.exists()
 
-        super().__init__(
+        val = cls.__model_new__(  # type: ignore
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:224]: this approach resolves having to type ignore every line
could add a class to inherit too, though I haven't been able to find any clever typing approaches so its basically just another flavor of untyped and likely adds an extra function call

File: python_modules/dagster/dagster/_model/decorator.py [hunk 2]
```diff
@@ -201,6 +213,12 @@ def _asdict(self):
         return as_dict(self)
 
 
+class IHaveNew:
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_model/decorator.py:216]: I think this is the cleanest option but i'm not sure the best name for this

Comment (schrockn) [reply python_modules/dagster/dagster/_model/decorator.py:216]: `IHaveCustomNew` probably more accurate?

Comment (alangenfeld) [reply python_modules/dagster/dagster/_model/decorator.py:216]: i would love to not repeat "new" , 

```
@dagster_model_with_new
class Foo(IHaveCustomNew):
```

```
@dagster_model_with
class Foo(CustomNew):
```

```
@dagster_model_override
class Foo(IHaveCustomNew):
```

**Comment by schrockn** [comment]
Oh I misunderstood.

Out of those options:

```python
@dagster_model_custom
class Foo(IHaveNew):
```

Seems good to me as it leaves us with the most option value.

**Review by schrockn — APPROVED** [review]

**Review by maximearmstrong — APPROVED** [review]

**Pull request merged**